substrate/executor/src/sandbox.rs

```diff
@@ -279,12 +263,50 @@ impl<'a, FE: SandboxCapabilities + Externals + 'a> Externals for GuestExternals<
 	}
 }
 
-struct SandboxInstance {
+fn with_guest_externals<FE, R, F>(
+	supervisor_externals: &mut FE,
+	sandbox_instance: &SandboxInstance,
+	state: u32,
+	f: F,
+) -> R
+where
+	FE: SandboxCapabilities + Externals,
+	F: FnOnce(&mut GuestExternals<FE>) -> R,
+{
+	let mut guest_externals = GuestExternals {
+		supervisor_externals,
+		sandbox_instance,
+		state,
+	};
+	f(&mut guest_externals)
+}
+
+pub struct SandboxInstance {
 	instance: ModuleRef,
 	dispatch_thunk: FuncRef,
 	guest_to_supervisor_mapping: GuestToSuperviserFunctionMapping,
 }
 
+impl SandboxInstance {
+	pub fn invoke<FE: SandboxCapabilities + Externals>(
```

Comment:
doc?

Reply:
Done!

Also added https://github.com/paritytech/polkadot/pull/153/commits/47b6a9f537484e224d77d3a0ec9225b9fb9cdc5a to the module